### ISSUE TITLE
How to control complexity of vertices?

### ISSUE THREAD

**Issue by mridvankeles** [SarahwXU/HiSup#26]
Hello,

As i can see we have c-iou metric which show us the complexity of our predictions. But im not sure how to control the number of vertex predictions. is it handled at the nms phase with topk parametre?

**Comment by SarahwXU** [comment]
Yes, the topk parameter impacts the number of vertex predictions. However, the value of topk is set to make sure that all proper junctions are detected. It barely influences the final complexity of building polygons. In fact, in our method, we suppose that the vertex detection is precise, which means the number of vertices matches the GT. There is no exact parameter for complex adjustment. If it has to be a parameter, the change of value of ''thres'' in the following function may slightly affect the complexity of polygons. https://github.com/SarahwXU/HiSup/blob/45db55433c9ef713dc52b7dd7f7d56e6645202eb/hisup/utils/polygon.py#L111  By the way, there is a small bug in the code of c-iou metric. I will fix it later. I am still sweating over the draft of my doctoral dissertation.
